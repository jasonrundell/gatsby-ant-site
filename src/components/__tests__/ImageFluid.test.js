import React from 'react'
import renderer from 'react-test-renderer'

import ImageFluid from '../ImageFluid'

describe('ImageFluid', () => {
  it('renders correctly', () => {
    const data = {
      childImageSharp: {
        fluid: {
          base64: '',
          aspectRatio: 1,
          src: '',
          srcSet: '',
          sizes: '',
        },
      },
    }
    const tree = renderer.create(<ImageFluid image={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
