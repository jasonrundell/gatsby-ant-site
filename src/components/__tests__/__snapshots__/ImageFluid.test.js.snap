// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageFluid renders correctly 1`] = `
<div
  className=" gatsby-image-wrapper"
  style={
    Object {
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={true}
    style={
      Object {
        "paddingBottom": "100%",
        "width": "100%",
      }
    }
  />
  
  <picture>
    <source
      sizes=""
      srcSet=""
    />
    <img
      alt=""
      loading="lazy"
      onLoad={[Function]}
      sizes=""
      src=""
      srcSet=""
      style={
        Object {
          "height": "100%",
          "left": 0,
          "objectFit": "cover",
          "objectPosition": "center",
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transition": "opacity 500ms",
          "width": "100%",
        }
      }
    />
  </picture>
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "<picture><source srcset=\\"\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
      }
    }
  />
</div>
`;
